diff --git a/node_modules/ember-model-validator/addon/initializers/model-locale.js b/node_modules/ember-model-validator/addon/initializers/model-locale.js
index cb5a6ef..78d3884 100644
--- a/node_modules/ember-model-validator/addon/initializers/model-locale.js
+++ b/node_modules/ember-model-validator/addon/initializers/model-locale.js
@@ -2,7 +2,7 @@ export function initialize() {
   let application = arguments[1] || arguments[0];
   let validatorDefaultLocale = application.get('validatorDefaultLocale');
   if (validatorDefaultLocale) {
-    application.register('validator:locale', validatorDefaultLocale, { instantiate: false });
+    application.register('validator:locale', { toString: () => validatorDefaultLocale }, { instantiate: false });
   }
 }
 
